synapse/storage/schema/main/delta/74/01membership_tables_event_stream_ordering.sql

```diff
@@ -0,0 +1,21 @@
+/* Copyright 2022 Beeper
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ALTER TABLE current_state_events ADD COLUMN event_stream_ordering BIGINT;
+ALTER TABLE local_current_membership ADD COLUMN event_stream_ordering BIGINT;
+ALTER TABLE room_memberships ADD COLUMN event_stream_ordering BIGINT;
```

Comment:
I think these can (and therefore should) be foreign key references on `events(stream_ordering)` ?

Reply:
We might also consider using triggers to enforce that the `stream_ordering` and `event_id` refer to the same row in `events` (see https://github.com/matrix-org/synapse/blob/bc1a1a9ffcc348f35b81b1056d47cedff221b27b/synapse/storage/schema/main/delta/68/04partial_state_rooms.sql for an example of this sort of thing).

Reply:
Hm any foreign key should be on `event_id` I would have thought, rather than a denormalise copy of data?

Definitely think the trigger is appropriate here to ensure consistency between `event_Id` & `event_stream_ordering`, personally this feels like the correct check over the foreign key..

Reply:
Added foreign keys in https://github.com/matrix-org/synapse/pull/15128/commits/8979abc928fd0410af0f03b3f95e818846eac432 and triggers for consistency in https://github.com/matrix-org/synapse/pull/15128/commits/36533a992ac4f636703ee0617b27481b244f9533.